### DISCUSSION TITLE
What is the difference between Transcrupt and Pyodide?

### DISCUSSION THREAD

**Discussion by ghost** [QQuick/Transcrypt#859] [General]
I'm looking for options to execute my python code in javascript and would like to know what I can do in order to best integrate the two languages. I was wondering if Transcrypt of Pyodide would be a better option, and would really appreciate any advice on what the nuances of either one are. Thanks!

**Comment by JennaSys** [comment]
The main difference between the two, is that Pyodide runs Python right in the browser via WebAssembly whereas Transcrypt transpiles Python into JavaScript preemptively as a separate step before it ever gets sent to the browser. The advantage of Transcrypt is that what is sent to the browser will be smaller and/or likely have better performance than running or compiling Python directly in the browser.

[This post by Anvil](https://anvil.works/blog/python-in-the-browser-talk) is getting a little dated as there are some new options available now (like PyScript), but it does a good job of explaining some of the differences in how they work. They all have valid use cases, it just depends on what your needs and priorities are.

**Reply by ghost** [reply]
Thank you! I'm much more interested in the concept of transcrypt as I think it stays truer to the webassembly paradigm. What version of python can we transcrypt into usable .js files?

**Reply by JennaSys** [reply]
Transcrypt has a hard dependency on Python 3.9 right now because it uses and is coded based the Python AST module. Using a different Python version where the AST might be different, can cause issues with how the JavaScript is generated.
